x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -268,6 +275,73 @@ public void testHighWatermarkCanBeExceededOnColdOrFrozenNode() throws Exception
         });
     }
 
+    public void testOvercommitInitializingSearchableSnapshotShards() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startNode(
+            Settings.builder()
+                .putList(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), DiscoveryNodeRole.DATA_HOT_NODE_ROLE.roleName())
+                .build()
+        );
+
+        InternalClusterInfoService masterInfoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(
+            ClusterInfoService.class
+        );
+        ClusterInfoServiceUtils.refresh(masterInfoService);
+
+        int nbIndices = createIndices();
+
+        assertAcked(
+            client().admin()
+                .cluster()
+                .preparePutRepository("repository")
+                .setType("mock")
+                .setSettings(Settings.builder().put("location", randomRepoPath()).build())
+        );
+        var mockRepository = (MockRepository) internalCluster().getCurrentMasterNodeInstance(RepositoriesService.class)
+            .repository("repository");
+        // Prevent searchable snapshot shards from quickly jumping from INITIALIZED to STARTED
+        mockRepository.setBlockOnceOnReadSnapshotInfoIfAlreadyBlocked();
+
+        createSnapshot("repository", "snapshot", nbIndices);
+
+        Map<String, Long> indicesStoresSizes = sizeOfShardsStores("index-*");
+        assertAcked(client().admin().indices().prepareDelete("index-*"));
+
+        String otherDataNode = internalCluster().startNode(
+            Settings.builder().put(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), DiscoveryNodeRole.DATA_COLD_NODE_ROLE.roleName()).build()
+        );
+        ensureStableCluster(3);
+
+        String otherDataNodeId = internalCluster().getInstance(NodeEnvironment.class, otherDataNode).nodeId();
+        logger.info(
+            "--> reducing disk size of node [{}/{}] so that all shards except one can fit on the node",
+            otherDataNode,
+            otherDataNodeId
+        );
+        String indexToSkip = randomFrom(indicesStoresSizes.keySet());
+        long totalSpace = indicesStoresSizes.entrySet()
+            .stream()
+            .filter(e -> e.getKey().equals(indexToSkip) == false)
+            .mapToLong(e -> e.getValue())
+            .sum() + WATERMARK_BYTES + 1024L;
+        getTestFileStore(otherDataNode).setTotalSpace(totalSpace);
+
+        logger.info("--> refreshing cluster info");
+        ClusterInfoServiceUtils.refresh(masterInfoService);
+        assertThat(
+            masterInfoService.getClusterInfo().getNodeMostAvailableDiskUsages().get(otherDataNodeId).getTotalBytes(),
+            equalTo(totalSpace)
+        );
+
+        mountIndices(indicesStoresSizes, "mounted-", "repository", "snapshot", FULL_COPY);
```

Comment:
We should validate that we do not allow allocating more shards that go over the disk threshold, right?

Reply:
Exactly! We check that we can't allocate N shards when there's a space only for N-1 shards

Reply:
Can we change the test to show that we indeed allocate `N-1` shards and the extra shards is not allocated?